src/__tests__/lib/utils/hooks/useImmutableTableQuery.test.tsx

```diff
@@ -0,0 +1,186 @@
+import { renderHook, act } from '@testing-library/react-hooks'
+import { cloneDeep } from 'lodash-es'
+import useImmutableTableQuery, {
+  UseImmutableTableQueryOptions,
+} from '../../../../lib/containers/useImmutableTableQuery'
+import * as DeepLinkingUtils from '../../../../lib/utils/functions/deepLinkingUtils'
+
+const options: UseImmutableTableQueryOptions = {
+  initQueryRequest: {
+    concreteType: 'org.sagebionetworks.repo.model.table.QueryBundleRequest',
+    entityId: 'syn123',
+    partMask: 1,
+    query: {
+      sql: 'SELECT * FROM syn123.3 WHERE "foo"=\'bar\'',
+    },
+  },
+}
+
+describe('useImmutableTableQuery tests', () => {
+  beforeEach(() => {
+    jest.clearAllMocks()
+  })
+
+  it('Returns the entity ID and version in the passed query', () => {
+    const { result } = renderHook(() => useImmutableTableQuery(options))
+    expect(result.current.entityId).toEqual('syn123')
+    expect(result.current.versionNumber).toEqual(3)
+  })
+
+  it('Updates the query using a new query object', () => {
+    const { result } = renderHook(() => useImmutableTableQuery(options))
+    expect(result.current.getLastQueryRequest()).toEqual(
+      options.initQueryRequest,
+    )
+
+    const newQuery = cloneDeep(options.initQueryRequest)
+    newQuery.query.sql = 'SELECT * FROM syn123.3 WHERE "foo"=\'baz\''
+
+    // Call under test
+    act(() => {
+      result.current.setQuery(newQuery)
+    })
+    expect(result.current.getLastQueryRequest()).toEqual(newQuery)
+    expect(result.current.getLastQueryRequest()).not.toBe(newQuery)
```

Comment:
Ohhh, this is testing that it is making a clone, right?  Could you add a comment?  It took me a minute of staring at lines 43 and 44 before coming to this conclusion :)

Reply:
Or remove since it's explicitly tested by the "Returns a deep clone of the initial request" below